alvr/client_openxr/src/extra_extensions/multimodal_input.rs

```diff
@@ -14,47 +14,104 @@ pub const META_DETACHED_CONTROLLERS_EXTENSION_NAME: &str = "XR_META_detached_con
+            let mut resume_simultaneous_hands_and_controllers_tracking_meta = None;
+            let _ = (session.instance().fp().get_instance_proc_addr)(
+                session.instance().as_raw(),
+                c"xrResumeSimultaneousHandsAndControllersTrackingMETA".as_ptr(),
+                &mut resume_simultaneous_hands_and_controllers_tracking_meta,
+            );
+
+            resume_simultaneous_hands_and_controllers_tracking_meta.map(|pfn| {
+                mem::transmute::<VoidFunction, ResumeSimultaneousHandsAndControllersTrackingMETA>(
+                    pfn,
+                )
+            })
+        }
+        .ok_or(sys::Result::ERROR_EXTENSION_NOT_PRESENT)?;
+
+        let pause_simultaneous_hands_and_controllers_tracking_meta = unsafe {
+            let mut pause_simultaneous_hands_and_controllers_tracking_meta = None;
+            let _ = (session.instance().fp().get_instance_proc_addr)(
+                session.instance().as_raw(),
+                c"xrPauseSimultaneousHandsAndControllersTrackingMETA".as_ptr(),
+                &mut pause_simultaneous_hands_and_controllers_tracking_meta,
+            );
+
+            pause_simultaneous_hands_and_controllers_tracking_meta.map(|pfn| {
+                mem::transmute::<VoidFunction, PauseSimultaneousHandsAndControllersTrackingMETA>(
+                    pfn,
+                )
+            })
+        }
+        .ok_or(sys::Result::ERROR_EXTENSION_NOT_PRESENT)?;
+
+        let resume_info = SimultaneousHandsAndControllersTrackingResumeInfoMETA {
+            ty: *TYPE_SIMULTANEOUS_HANDS_AND_CONTROLLERS_TRACKING_RESUME_INFO_META,
+            next: ptr::null(),
+        };
+        unsafe {
+            super::xr_res(resume_simultaneous_hands_and_controllers_tracking_meta(
+                session.as_raw(),
+                &resume_info,
+            ))?;
+        }
+
+        Ok(Self {
+            session,
+            pause_simultaneous_hands_and_controllers_tracking_meta,
         })
     }
-    .ok_or(sys::Result::ERROR_EXTENSION_NOT_PRESENT)?;
-
-    let resume_info = SimultaneousHandsAndControllersTrackingResumeInfoMETA {
-        ty: *TYPE_SIMULTANEOUS_HANDS_AND_CONTROLLERS_TRACKING_RESUME_INFO_META,
-        next: ptr::null(),
-    };
-
-    unsafe {
-        super::xr_res(resume_simultaneous_hands_and_controllers_tracking_meta(
-            session.as_raw(),
-            &resume_info,
-        ))?;
-    }
+}
 
-    Ok(())
+impl Drop for MultimodalHandsHandle {
+    fn drop(&mut self) {
+        let pause_info = SimultaneousHandsAndControllersTrackingPauseInfoMETA {
+            ty: *TYPE_SIMULTANEOUS_HANDS_AND_CONTROLLERS_TRACKING_PAUSE_INFO_META,
+            next: ptr::null(),
+        };
+        unsafe {
+            super::xr_res((self
+                .pause_simultaneous_hands_and_controllers_tracking_meta)(
+                self.session.as_raw(),
+                &pause_info,
+            ))
+            .ok();
+        }
+    }
```

Comment:
Also don't think implementing drop is worth it, it seems more advantageous to just remain in the last state.

Reply:
Yeah so this means going back to global state...